Develop

krauson opened the pull request.

krauson committed:
update

krauson committed:
go to date

krauson committed:
go to date files

krauson committed:
Merge pull request #1 from PythonFreeCourse/develop

Develop

krauson committed:
move go_to_date into app dir

krauson committed:
test

krauson committed:
delete

krauson committed:
Merge branch 'develop' of https://github.com/PythonFreeCourse/calendar into feature/go_to_date

krauson committed:
Merge branch 'develop' of https://github.com/krauson/calendar into develop

Gonzom committed:
feat: add i18n support (#115)

* feat: add i18n support

This feature includes the core functions added to /internal, English and Hebrew json files, removal of text strings from the html files, updates to the routes and model updates. Also included are tests. Please note, that this feature is waiting on user registration for final hooks.

By: Gonny <1@1>

yammesicka committed:
Revert "feat: add i18n support (#115)" (#161)

This reverts commit 779de2d0735b00448c49c68cfc394c78b82c5421.

krauson committed:
Merge branch 'develop' of https://github.com/PythonFreeCourse/calendar into develop

krauson committed:
Merge branch 'develop' of https://github.com/PythonFreeCourse/calendar into develop

krauson committed:
Merge branch 'develop' of https://github.com/PythonFreeCourse/calendar into develop

krauson committed:
Merge branch 'feature/go_to_date' into develop

krauson committed:
adding go_to_date

krauson committed:
Merge branch 'feature/go_to_date' into develop

krauson committed:
check reverting

yammesicka commented:
Please make sure you give better names for the PR, and better commit messages :)

yammesicka commented on app/config.py:
```diff
@@ -0,0 +1,35 @@
+import os
```
Please do not upload the config.py file

yammesicka commented on app/go_to_date/main.py:
```diff
@@ -0,0 +1,18 @@
+from fastapi import FastAPI, Request
```
Please choose an existing directory

krauson replied:
Didn't quite understand what did you mean?
what should I change?

yammesicka replied:
The path of the file should be app/(EXISTING_DIRECTORY_HERE)/main.py, not `go_to_date`

krauson replied:
I still don't understand, should I put my feature's files directly in the "app" directory?
don't really understand what you mean by existing directory?

yammesicka replied:
There are many directories that are already exists: `routers`, `internal` etc.

Choose one of them which fits the best, according to the schema described [here](https://fastapi.tiangolo.com/tutorial/bigger-applications/), and move this file to there.

Don't create your own directory for each feature :)

krauson replied:
Ok, so if I understand it right I need to put my python file at the internal folder (and maybe change its name to go_to_date), and the content of static folder I should put in the static folder app/static, and the same goes for the templates folder?

yammesicka commented on app/go_to_date/static/styles.css:
```diff
@@ -0,0 +1,70 @@
+  cursor: pointer;
+  opacity: 0.8;
+  position: fixed;
+  bottom: 23px;
```
Do not work with `px`, work with em or % instead :)

yammesicka commented on app/go_to_date/todo.py:
```diff
@@ -0,0 +1,2 @@
+# 1.button
```
Remove this file

yammesicka commented on app/revert_test.txt:
```diff
@@ -0,0 +1 @@
+"hello world" 
```
Remove this file, and dev.db

yammesicka commented on go_to_date/main3.py:
```diff
@@ -0,0 +1,19 @@
+from fastapi import FastAPI, Request, Form
```
Please make sure you send the commit after you check all the files are relevent and ready for check.

krauson committed:
delete config and revert_test

krauson committed:
remove todo file

krauson committed:
Merge branch 'develop' of https://github.com/PythonFreeCourse/calendar into develop

yammesicka committed:
Update our production site. (#209)

* update ORM and added export for an event

* update ORM and added export for an event

* Added in app shareable events

* docs: add type annotation and fix folder structure

* docs: add type annotation and fix folder structure

* docs: fix documentation

* feat: get weather forecast for date and location

* feat: get weather forecast for date and location

* add: tests

* set up commit

* add: timezone support

* add: session management

* fix bug

* split conftest file

* split conftest file

* move "utils" folder into "internal" folder

* working day view with jinja2 template

* with tests for pytest

* feat: get weather forecast - fixes according to requested changes.

* change file structure

* feat: get weather forecast - fixes according to requested changes.

* first functioning day-view html, css and router

* feat: Basic responsive calender day view page

Added dayview template with css and router.
An tempreroy event class with,
 processing function for the day view to work.

* feat: Basic responsive calender day view page

Added dayview template with css and router.
An tempreroy event class with,
 processing function for the day view to work.
fixed after taking notes.

* fix lint now for sure

* more lint

* feat: get weather forecast - fix requirements.txt

* feat: enable invited users to view events

* feat: flake8 changes

* fix: requirements bug

* fix: requirements bug

* feat: flake8 changes

* add: tests

* feat: flake8 changes

* add: tests

* feat: flake8 changes

* edit: file structure

* edit: file structure

* feat: get weather forecast - fix changes & add cache support

* feat: get weather forecast - fix changes & add cache support

* feat: get weather forecast - fix changes & add cache support

* feat: add route tests

* feat: get weather forecast - fix changes & add cache support

* feat: get weather forecast - fix changes & add cache support

* fix: test bug

* remove: config.py

* after all notes and started use the orm

* fix: type annotation

* feat: get weather forecast - add API mocking

* feat: get weather forecast - add API mocking

* feat: get weather forecast - add API mocking

* feat: get weather forecast - add API mocking

* fixed: changed "size" var in html and and more acuurate typing

* deleted some testing lines

* add requirements missing

* add: minor changes

* feat: flake8 changes

* @hadaskedar2020

* feat: get weather forecast - add API mocking

* feat: weather forecast - add API mocking & improve coverage

* feat: weather forecast - add API mocking & improve coverage

* feat: weather forecast - add API mocking & improve coverage

* feat: weather forecast - add API mocking & improve coverage

* feat: weather forecast - add API mocking & improve coverage

* fixed: statements of event select more accurate

* new test and lint fix

* added more contants abd more notes taken

* feat: weather forecast - move feat to internal

* Delete Scripts directory

* update constants

* more notes

* forgot white space

* feat: gets a link to the suitable image of a given event content. (#70)

* feat: gets a link to the suitable image of a given event content.

* Add basic update event (#100)

* Add basic update event

* Added search feature (#98)

* Added search feature

* Add telegram client (#111)

Add telegram client

* feat: Add a feature for sharing information with a WhatsApp account (#83)

* feat: Add a feature to share to a WhatsApp account

* Feature/logging system (#112)

* feat: first logging commit

Co-authored-by: Liad Noam <LiadN@Taldor.corp>

* feat: add a daily inspirational quote (#128)

* feat: add a daily inspirational quote

* Feature/delete event (#139)

* basic delete event

* Add an event delete function

* Fix flake8

* Deleting an unnecessary model

* Extract lines from try

* fix: modify psycopg2 dependency (#164)

* Bugfix/logging system (#158)

* Changed logging config to be received by separate parameters for simplicity, removed option to receive configuration as a dict, added .vscode to gitignore

* removed logging_config file which is no longer needed

* Fixing merge conflicts - missing whitespace on config.py.example

Co-authored-by: Liad Noam <LiadN@Taldor.corp>

* Added checking for zoom link and saving the event (#122)

* Added checking for zoom link and saving the event

* Feature/import to calendar (#119)

* feat: import file to calendar

* docs: adding community standard documents (#173)

* Feat: Event View Backend (#159)

* Moved the route to top of file

* Feature/send email (#88)

* Added files for sending invitations by email

This includes
- templates
- routes and internal code
- configuration
- tests

Co-authored-by: leddest <46251307+leddest@users.noreply.github.com>
Co-authored-by: Ellen <ellenc345@gmail.com>

* Feature/translation - add an functionality to translate event data etc. (#167)

* translation feature stuff

* Update event - Improving the code (#138)

* Add basic update event

* Support categories for events (#137)

* Support categories for events

* Style/grid (#106)

* Hello Wrold

* Creating Html Template

* Calendar dates calculation intigrating with html, HTML\CSS improvments.

* Tests for calender_grid.py, changing function according to currections, fixing bugs

* linting bug fixing

* before merge with develop

* Creating a Day object and days subclasses, changing all function and tests according to the new objects, Changing front to get information from Day objects, Re-arranging  calendar.html grid stracture to be render by weeks, adding js functionality for day view section, adding css effects on daily event display

* fix lintings

* fix lintings

* tests next week scroll

* js updates

* Calendar Scrolling with javascript

* Fix lintings

* Fix lintings

* Calendar infinit scrolling ducplicates weeks bug fixed

* seetings global parameters for calendar.py

* Js - changing to fetch, removing jquery, fixing hint tpying, adding Week object, changing tests

* Front end bug fixing

* Fix linting

* Fix lintings

* Fix lintings

* fixing syntax

* Get user local date and time when enter calendar.

* Fix lintings

* Fix lintings

* Fix lintings

* Bugs fixing

* Bugs fixing

* Bugs fixing

* Bugs fixing

* JS alerts removal

* Fix Lintings

* Js syntext fixing

* Bugs fixing

* Bugs fixing

* Js fixing varibales and adding div element to request.

* Develop Update

* Js - eliminate global parameters

* Linting fixing

* Fix Lintings

* Fix flake8

* syntac fixing

* Feature/translations workflow (#150)

* feat: add translations workflow

* fix(translations-workflow): fix doc

* fix(translations-workflow): fixed code and added automatic commit

Co-authored-by: Gonny <1@1>

* Revert "Feature/translations workflow (#150)" (#208)

This reverts commit 2e6a2c2cb5eac79d5dadef321793c0b3518ff7a6.

* feat: Add zodiac signs to the day & month view + refactor json data loader (#155)

* feat: Add zodiac signs to the day/month view +

* Change calendar link from '#' to '/' to get the home page (#194)

when clicking on the logo

* Style/grid (#200)

* Js scrolling duplication days fixed

Co-authored-by: Idan Pelled <pelled.idan@gmail.com>
Co-authored-by: Hadas Kedar <hadasit@gmail.com>
Co-authored-by: Sagi Zaid Or <sagi@eretz.org.il>
Co-authored-by: sagizaidor <71097154+sagizaidor@users.noreply.github.com>
Co-authored-by: hadaskedar2020 <70915466+hadaskedar2020@users.noreply.github.com>
Co-authored-by: nir9696 <35953934+nir9696@users.noreply.github.com>
Co-authored-by: efratush <70986341+efratush@users.noreply.github.com>
Co-authored-by: advaa123 <64307569+advaa123@users.noreply.github.com>
Co-authored-by: leddest <46251307+leddest@users.noreply.github.com>
Co-authored-by: nelliei <71073199+nelliei@users.noreply.github.com>
Co-authored-by: Liad-n <52039539+Liad-n@users.noreply.github.com>
Co-authored-by: Liad Noam <LiadN@Taldor.corp>
Co-authored-by: zohary89 <71074080+zohary89@users.noreply.github.com>
Co-authored-by: Ron Huberfeld <32178925+ron-huberfeld@users.noreply.github.com>
Co-authored-by: Ap1234567 <58956386+Ap1234567@users.noreply.github.com>
Co-authored-by: Nir-P <70997974+Nir-P@users.noreply.github.com>
Co-authored-by: Ode <odechann@gmail.com>
Co-authored-by: Anna Shtirberg <71028276+annashtirberg@users.noreply.github.com>
Co-authored-by: Ellen <ellenc345@gmail.com>
Co-authored-by: ivarshav <ivarshav@users.noreply.github.com>
Co-authored-by: Aviad <38867584+aviadamar@users.noreply.github.com>
Co-authored-by: Gonzom <gonnym@gmail.com>
Co-authored-by: Gonny <1@1>
Co-authored-by: YairEn <71101056+YairEn@users.noreply.github.com>

krauson committed:
Merge branch 'develop' of https://github.com/PythonFreeCourse/calendar into develop

krauson committed:
change and delete files Yam told me to change

krauson committed:
change README file

yammesicka commented on README.md:
```diff
@@ -35,7 +35,7 @@ uvicorn app.main:app --reload
 source venv/bin/activate
 pip install -r requirements.txt
-cp app/config.py.example app/config.py
+cp app/config.py.example app/configuration.py
```
Change back to config.py

yammesicka commented on app/go_to_date/static/popup.js:
```diff
@@ -0,0 +1,7 @@
+function openForm() {
+    document.getElementById("myForm").style.display = "block";
```
Use CSS and create class which the property `{ display: block }`, then use `toggleClass` to add/remove this class from `openForm`.

Don't edit the style directly from the JavaScript, as we try not to mix logic with styling.

krauson replied:
I got a bit confused,see toggleClass belongs to jquery library which uses a different click function (not addEventListener), so should I use something like that?:
$(document).ready(function(){
    $("button").click(function(){
      $("p").toggleClass("main");

yammesicka replied:
Yeah sure. I meant you should use `element.classList.toggle`

yammesicka commented on app/go_to_date/templates/index.html:
```diff
@@ -0,0 +1,26 @@
+    <title>Document</title>
+</head>
+<body>
+    <button class="open-button" onclick="openForm()">Go to a specific date</button>
```
Use JavaScript's `addEventListener`, don't use inline `onclick` event

yammesicka replied:
Remove the `openForm` (and don't mark as resolved before you actually resolve it)

yammesicka commented:
Great progress! I've added few more insights :)
